transform/app/worker.py

```diff
@@ -1,63 +1,22 @@
 # pylint: disable=invalid-name, logging-fstring-interpolation
 """Celery worker. Responsibilities: get, transform, upload data"""
 import os
-import json
-import logging
 from celery import Celery
-import app.transform.transformers as trans
-from app.transform.utils.loader import (
-    load_env_vars,
-    load_request_data,
-)
-from app.transform.schemas.properties.env import (
-    ALL_COLLECTION,
-    GUIDELINE,
-    INPUT_SCHEMA,
-)
-from app.transform.utils.send import send_json_string_to_solr
-from app.services.spark.config import apply_spark_conf
-from app.services.solr.delete import delete_data_by_type
 
+script_directory = os.path.dirname((os.path.abspath(__file__)))
+
+tasks_module_path = os.path.join(script_directory, "tasks")
 
-logger = logging.getLogger(__name__)
-celery = Celery(__name__)
+tasks_modules = [
```

Comment:
seems like not the best way to do it, but maybe I am wrong. It was a recommended way when you have done research?

Reply:
The basic way is to use the include argument, e.g. `celery = Celery(__name__, include=['app.tasks.batch', 'app.tasks.create_collections'])`. I could create a variable `modules_to_include = [...]` at the beginning of the file, the second option is `celery.autodiscover_tasks()`, which works best if you stick to the naming convention of task files, e.g. `tasks.py` and put them in multiple modules. The third option is dynamic task detection, which is what I was trying to achieve. At least these are the ways I could find. I chose the first way because, due to our convention the second way with autodiscover_tasks would be longer than the first as the project grows.